src/anonymousUserTracking/tests/nestedTesting.test.ts

```diff
@@ -92,4 +92,158 @@ describe('nestedTesting', () => {
     const result = checker.getMatchedCriteria(JSON.stringify(NESTED_CRITERIA));
     expect(result).toEqual(null);
   });
+
+  it('should return criteriaId 425 (Multi level Nested field criteria)', () => {
+    (localStorage.getItem as jest.Mock).mockImplementation((key) => {
+      if (key === SHARED_PREFS_EVENT_LIST_KEY) {
+        return JSON.stringify([
+          {
+            eventName: 'button-clicked',
```

Comment:
Can we make sure that the data that is passed in for the test matching the updateCart API parameters: https://api.iterable.com/api/docs#commerce_updateCart

We should have 2 separate tests. One for the eventName "button-clicked" and one for eventName "updateCart".

Reply:
This changes are not relatable to the "updateCart" so removed "updateCart" from the criteria,

All changes are related to "customEvent" only